temp_pg_db.py

```diff
@@ -49,31 +73,40 @@ def __init__(self, databases=None, verbosity=0):
             os.mkdir(self.pg_data_dir)
             self.pg_socket_dir = os.path.join(self.pg_temp_dir, 'socket')
             os.mkdir(self.pg_socket_dir)
-            print "Creating temp PG server...",
+            printf("Creating temp PG server...")
             sys.stdout.flush()
-            rc = subprocess.call(['initdb', self.pg_data_dir], stdout=stdout, stderr=stderr) == 0
+            rc = subprocess.call([initdb, self.pg_data_dir],
+                                 stdout=stdout, stderr=stderr) == 0
             if not rc:
                 raise PGSetupError("Couldn't initialize temp PG data dir")
             self.pg_process = subprocess.Popen(
-                ['postgres', '-F', '-T', '-D', self.pg_data_dir, '-k', self.pg_socket_dir],
+                [postgres, '-F', '-T',
+                 '-D', self.pg_data_dir,
+                 '-k', self.pg_socket_dir,
+                 '-h', ''],
                 stdout=stdout, stderr=stderr)
+
             # test connection
-            for i in xrange(5):
-                time.sleep(1)
-                rc = subprocess.call(['psql', '-d', 'postgres', '-h', self.pg_socket_dir, '-c', "\dt"],
-                    stdout=stdout, stderr=stderr) == 0
+            for i in range(retry):
+                time.sleep(tincr)
+                rc = subprocess.call([psql, '-d', 'postgres',
+                                      '-h', self.pg_socket_dir,
+                                      '-c', "\dt"],
+                                     stdout=stdout, stderr=stderr) == 0
                 if rc:
                     break
             else:
                 raise PGSetupError("Couldn't start PG server")
             rc = True
             for db in databases:
-                rc = rc and subprocess.call(['psql', '-d', 'postgres', '-h', self.pg_socket_dir, '-c',
-                                             "create database %s;" % db], stdout=stdout, stderr=stderr) == 0
+                rc = rc and subprocess.call([psql, '-d', 'postgres',
+                                             '-h', self.pg_socket_dir,
+                                             '-c', "create database %s;" % db],
+                                            stdout=stdout, stderr=stderr) == 0
             if not rc:
                 raise PGSetupError("Couldn't create databases")
-            print "done"
-            print "(Connect on: `psql -h %s`)" % self.pg_socket_dir
+            print("done")
+            print("(Connect on: `psql -h %s`)" % self.pg_socket_dir)
```

Comment:
In python2 (and python1?) the `(expr)` form simplifies to `expr` when it does not contain commas.  If it contains commas then it'll become a tuple.  Since we're passing a single string then then `(str)` becomes `str` when it's consumed by the python2 `print` keyword.

Sending a single string through is the lowest common denominator that works across 2+3.